magento/magento2
# Extend Observer declaration in events.xml by new attribute ifconfig

### Description (*)
Extension of Observer declaration in events.xml by new optional attribute **ifconfig** with appropriate handling of it.
Similar to usage of **ifconfig** in layout's xml files - it executes (or not) Observer depending on boolean value of given config path. The purpose is to condition Observer's execution on config paths representing module's state - usually <module/general/active> or <module/general/enabled> etc.

### Manual testing scenarios (*)

1. Add ifconfig attribute to Observer A declaration in events.xml file where its value will be config path and dispatch event to trigger Observer. 
a) When config path value equals to **true** - Observer should execute.
b) When config path value equals to **false** - Observer should **not** execute
c) When config does not exist (path does not represent any actual value) - Observer should **not** execute
2. Add ifconfig attribute to Observer A declaration in events.xml file where its value is empty string "" and dispatch event to trigger Observer. Observer should execute. 


### Contribution checklist (*)



### Resolved issues:
1. [x] resolves magento/magento2#38514: Extend Observer declaration in events.xml by new attribute ifconfig
